lib/membrane_h264_ffmpeg/encoder.ex

```diff
@@ -156,6 +157,19 @@ defmodule Membrane.H264.FFmpeg.Encoder do
     {:ok, %{state | encoder_ref: nil}}
   end
 
+  defp flush_encoder_if_exists(nil) do
```

Comment:
You could make `flush_encoder_if_exists` take `state` as an argument, and then match if encoder is `nil`. This would simplify the code, you are matching `encoder_ref` ~3 times.